reflect.go

```diff
@@ -464,44 +473,43 @@ func recordedObjectString(obj types.Object) objectString {
 	return pkg.Path() + "." + obj.Name()
 }
 
-// reflectedObjectString returns the obfucated name of a types.Object,
+// obfuscatedObjectName returns the obfucated name of a types.Object,
 // parent is needed to correctly get the obfucated name of struct fields
-func reflectedObjectString(obj types.Object, parent *types.Struct) string {
-	if obj == nil {
-		return ""
-	}
```

Comment:
I think that the obfuscated name of a nil obj is `""` is fine. I don't see a benefit of the callers having to make sure obj is non-nil.

Reply:
I don't think this is a big deal either way, but I personally disagree - a nil obj is an edge case that we should break out of as soon as reasonable, rather than keep going down the logic. All other callers already ensure that obj is non-nil, it was just two that were careless. If any code path ends up with a nil obj, that's possibly a bug that we need to investigate and fix - rather than silently assuming that the result is an empty string.